### ISSUE TITLE
[FR] Consider migration merges and improved tests.

### ISSUE THREAD

**Issue by miggland** [inventree/InvenTree#4963]
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find a similar feature request

### Problem statement

A developer (me) messed up a set of migration files, which caused big problems.

My blunder is explained by this:

> Basically, if you have to mess around with the order of migrations, you should never change one which has already been run! I should have checked this more carefully before merging in. The CI checks passed because they operate on a clean database, and the tables here had not already been created.

Relevant PRs:
https://github.com/inventree/InvenTree/pull/4898
https://github.com/inventree/InvenTree/pull/4961

### Suggested solution

1. Django has built-in tools which can handle some migration conflicts (`makemigration --merge`). These should probably be the go-to choice.
2. CI checks could be thought of.

For the first issue, there's a few things which could be improved on:
- Describe what to do, and what to avoid, when considering migrations in CONTRIBUTING.md
- Make --merge accessible through invoke, with some tips.

For the 2nd part, a suggestion is:
> 
> 1. New PR is created
> 2. As part of CI, old database is fetched which is running on top of current master
> 3. If database migrations apply, all is OK
> 4. Once PR is merged, triggers a workflow to update the database with the new migrations

### Describe alternatives you've considered

Maybe not all of the above needs to be implemented, maybe there's a better solution.

### Do you want to develop this?

- [ ] I want to develop this.

**Comment by miggland** [comment]
There's some more information about merge conflicts on this page:
https://www.algotech.solutions/blog/python/django-migrations-and-how-to-manage-conflicts/

**Comment by matmair** [comment]
We already have section on migrations, maybe this needs to be update: https://github.com/inventree/InvenTree/blob/master/CONTRIBUTING.md#migration-files
